File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        originalN = n
        sum_ = 0
        prod_ = 1
        while n > 0:
            digit = n % 10
            sum_+=digit
            prod_*=digit
            n = n//10

        return originalN % (sum_ + prod_) ==0    

        

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int originalN = n;
        int prod = 1;
        int sum = 0;
        while (n > 0){
            int digit = n % 10;
            prod = prod * digit;
            sum = sum + digit;
            n = n/10;
        }

        return originalN % (sum + prod)==0;
    }
}